worker/lease/manager.go

```diff
@@ -207,12 +207,21 @@ func (manager *Manager) retryingClaim(claim claim) {
 
 	if success {
```

Comment:
I think this block can be made a little more easy to follow. With the returns, we don't need the else if etc:

```
if lease.IsTimeout(err) {
    claim.respond(lease.ErrTimeout)
    return
}
if err == nil {
    if !success {
        claim.respond(lease.ErrClaimDenied)
        return
    }
    // Do not return here - this is pretty subtle .... etc
    claim.respond(nil)
}
```

Reply:
That's much nicer, thanks.